src/wallet/rpcdump.cpp

```diff
@@ -74,19 +77,166 @@ std::string DecodeDumpString(const std::string &str) {
     return ret.str();
 }
 
+bool SweepPrivKey(CBitcoinSecret key, CBitcoinAddress destAddress) {
+    // Get the private key to be swept and validate it
+    CKey dirtyKey = key.GetKey();
+    if (!dirtyKey.IsValid())
+        return false;
+
+    // Generate public key and verify it
+    CPubKey pubkey = dirtyKey.GetPubKey();
+    if (!dirtyKey.VerifyPubKey(pubkey))
+        return false;
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    // Get outputs from wallet
```

Comment:
Why would you sweep it, if it's in the wallet (and thereby trusted to be exclusively yours)?